logbook-servlet/src/main/java/org/zalando/logbook/servlet/LocalResponse.java

```diff
@@ -106,42 +114,54 @@ public PrintWriter getWriter() throws IOException {
         return body;
     }
 
-    private final class TeeServletOutputStream extends ServletOutputStream {
+    private static final class TeeServletOutputStream extends ServletOutputStream {
 
-        private final OutputStream original;
+        private final DataOutput output;
+        private final LazyStream original;
 
-        private TeeServletOutputStream() throws IOException {
-            this.original = LocalResponse.super.getOutputStream();
+        private TeeServletOutputStream(final DataOutput output,
+                final LazyStream original) {
```

Comment:
Or maybe just introducing flag `withoutBody` and postponing creation of `TeeServletOutputStream` until `LocalResponse.getOutputStream` is called would be even better way and will eliminate all this unobvious lazy stuff